File: Modules/module_sample/src/main/java/com/example/william/my/module/sample/activity/MvvmActivity.java
package com.example.william.my.module.sample.activity;

import android.os.Bundle;

import com.alibaba.android.arouter.facade.annotation.Route;
import com.blankj.utilcode.util.FragmentUtils;
import com.example.william.my.library.base.BaseActivity;
import com.example.william.my.module.router.ARouterPath;
import com.example.william.my.module.sample.R;
import com.example.william.my.module.sample.fragment.MvvmFragment;

@Route(path = ARouterPath.Sample.Sample_MVVM)
public class MvvmActivity extends BaseActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.sample_layout_fragment);

        // Add product list fragment if this is first creation
        MvvmFragment mvvmFragment = (MvvmFragment) getSupportFragmentManager()
                .findFragmentById(R.id.contentFrame);

        if (mvvmFragment == null) {
            mvvmFragment = MvvmFragment.newInstance();
            FragmentUtils.add(getSupportFragmentManager(), mvvmFragment, R.id.contentFrame);
        }
    }
}

File: modules/module_sample/src/main/kotlin/com/example/william/my/module/sample/activity/MvvmActivity.kt
package com.example.william.my.module.sample.activity

import androidx.fragment.app.Fragment
import com.alibaba.android.arouter.facade.annotation.Route
import com.example.william.my.library.base.BaseFragmentActivity
import com.example.william.my.module.router.ARouterPath
import com.example.william.my.module.sample.fragment.MvvmFragment

@Route(path = ARouterPath.Sample.Sample_MVVM)
class MvvmActivity : BaseFragmentActivity() {

    override fun setFragment(): Fragment {
        return MvvmFragment()
    }

}